download_weather_streaming.py

```
@@ -0,0 +1,308 @@
+    end_date: Optional[datetime] = None,
+    max_locations: Optional[int] = None,
+    max_concurrent: int = 5,
+    output_dir: Optional[Path] = None
+) -> Path:
+    """Download weather data with streaming to avoid memory issues"""
+    
+    registry = get_registry()
+    registry.auto_discover()
+    
+    if source not in registry.list_plugins():
+        raise ValueError(f"Unknown data source: {source}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)